src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java

```diff
@@ -126,12 +127,29 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
                 );
                 String defaultsFilePath = DefaultUseCases.getDefaultsFileByUseCaseName(useCase);
                 useCaseDefaultsMap = ParseUtils.parseJsonFileToStringToStringMap("/" + defaultsFilePath);
-
-                if (request.hasContent()) {
+                List<String> requiredParams = DefaultUseCases.getRequiredParamsByUseCaseName(useCase);
+
+                if (request.hasContent() == false && requiredParams.size() != 0) {
+                    throw new FlowFrameworkException(
+                        "Missing the following required parameters for use case [" + useCase + "] : " + requiredParams.toString(),
+                        RestStatus.BAD_REQUEST
+                    );
+                } else {
                     try {
                         XContentParser parser = request.contentParser();
                         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
                         Map<String, Object> userDefaults = ParseUtils.parseStringToObjectMap(parser);
+
+                        // Validate user defaults key set
+                        Set<String> userDefaultKeys = userDefaults.keySet();
+                        if (!userDefaultKeys.containsAll(requiredParams)) {
+                            requiredParams.removeAll(userDefaultKeys);
+                            throw new FlowFrameworkException(
+                                "Missing the following required parameters for use case [" + useCase + "] : " + requiredParams.toString(),
+                                RestStatus.BAD_REQUEST
+                            );
+                        }
+
```

Comment:
Rather removing from `requiredParams` List. We can do

```java
Set<String> userDefaultKeys = userDefaults.keySet();
Set<String> missingParams = new HashSet<>(requiredParams);
missingParams.removeAll(userDefaultKeys);

if (!missingParams.isEmpty()) {
    throw new FlowFrameworkException(
        "Missing the following required parameters for use case [" + useCase + "] : " + missingParams,
        RestStatus.BAD_REQUEST
    );
}

```

Reply:
`DefaultUseCases.getRequiredParamsByUseCaseName()` just returns a copy of the required param list, so we're not modifying the original list.